frontend/src/main/web/lib/components/glossary/DataTable.jsx

```diff
@@ -0,0 +1,486 @@
+import React, {PureRenderMixin} from 'react/addons';
+import Actions from '../../actions/GlossaryActions';
+import {Table, Column} from 'fixed-data-table';
+import StringUtils from '../../utils/StringUtils'
+import InputCell from './InputCell';
+import LoadingCell from './LoadingCell'
+import ActionCell from './ActionCell'
+import SourceActionCell from './SourceActionCell'
+import ColumnHeader from './ColumnHeader'
+import _ from 'lodash';
+
+var DataTable = React.createClass({
+  TIMEOUT: 400,
+
+  NO_ROW: -1,
+
+  CONTENT_HASH_INDEX: 0,
+
+  ENTRY: {
+    SRC: {
+      col: 1,
+      field: 'srcTerm.content',
+      sort_field: 'src_content'
+    },
+    TRANS: {
+      col: 2,
+      field: 'transTerm.content',
+      sort_field: 'trans_content'
+    },
+    POS: {
+      col: 3,
+      field: 'pos',
+      sort_field: 'part_of_speech'
+    },
+    DESC: {
+      col: 4,
+      field: 'description',
+      sort_field: 'desc'
+    },
+    TRANS_COUNT: {
+      col: 5,
+      field: 'termsCount',
+      sort_field: 'trans_count'
+    }
+  },
+  CELL_HEIGHT: 48,
+
+  propTypes: {
+    glossaryData: React.PropTypes.object.isRequired,
+    glossaryHash: React.PropTypes.arrayOf(
+      React.PropTypes.arrayOf(React.PropTypes.string)
+    ),
+    canAddNewEntry: React.PropTypes.bool.isRequired,
+    canUpdateEntry: React.PropTypes.bool.isRequired,
+    user: React.PropTypes.shape({
+      username: React.PropTypes.string,
+      email: React.PropTypes.string,
+      name: React.PropTypes.string,
+      imageUrl: React.PropTypes.string,
+      languageTeams: React.PropTypes.string
+    }),
+    srcLocale: React.PropTypes.shape({
+      locale: React.PropTypes.shape({
+        localeId: React.PropTypes.string.isRequired,
+        displayName: React.PropTypes.string.isRequired,
+        alias: React.PropTypes.string.isRequired
+      }).isRequired,
+      numberOfTerms: React.PropTypes.number.isRequired
+    }),
+    selectedTransLocale: React.PropTypes.string,
+    totalCount: React.PropTypes.number.isRequired,
+    focusedRow: React.PropTypes.shape({
+      contentHash: React.PropTypes.string,
+      rowIndex: React.PropTypes.number
+    })
+  },
+
+  getInitialState: function () {
+    var top = 246; //top height for banner if can't get height from dom
+    return {
+      tbl_width: this._getWidth(),
+      tbl_height: this._getHeight(top),
+      row_height: this.CELL_HEIGHT,
+      header_height: this.CELL_HEIGHT,
+      hoveredRow: -1
+    }
+  },
+
+  /**
+   *
+   * @param fixedTop - gets the height from the DOM if this is not specified
```

Comment:
- [ ] This tells me the default if I do not specify `fixedTop`, but I still wonder what the value of fixedTop is used for, and what type is expected.

It looks like maybe it expects a `number` and it looks like it is how you tell it where the top of the element is, and if you don't specify where the top is it will look for it in the DOM.